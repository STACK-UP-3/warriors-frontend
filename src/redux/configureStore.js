import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
