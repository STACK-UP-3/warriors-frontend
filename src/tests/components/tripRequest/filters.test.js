import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import Filters from '../../../components/tripRequests/viewTripRequests/filters';
import configureStore from '../../../redux/configureStore';

describe('*************** Testing the Filters  component ***************',()=>{

    const store = configureStore();
    let statusChange,locationMock,historyMock, wrapper;

    beforeEach(()=>{
        statusChange = jest.fn();
        locationMock = {
                pathname: '/viewtrips/accepted',
                search: '',
                state: '',
                hash: ''
        };
        
        historyMock ={ 
            replace: jest.fn(),
            length: 0,
            location: { 
                pathname: '',
                search: '',
                state: '',
                hash: ''
            },
            action: 'REPLACE',
            push: jest.fn(),
            go: jest.fn(),
            goBack: jest.fn(),
            goForward: jest.fn(),
            block: jest.fn(),
            listen: jest.fn(),
            createHref: jest.fn()
            };
        
        wrapper =  mount(
            <Provider store={ store }> 
                <Router location={locationMock} history={historyMock}>
                    <Filters statusChange={ statusChange }  />
                </Router>
            </Provider>
        );
    });
    
    it('Should render the filters component correctly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
});
