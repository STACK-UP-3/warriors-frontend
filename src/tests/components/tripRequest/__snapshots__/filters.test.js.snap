// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the Filters  component *************** Should render the filters component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 0,
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "hash": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/viewtrips/accepted",
        "search": "",
        "state": "",
      }
    }
  >
    <_default
      statusChange={[MockFunction]}
    >
      <div
        className="filtersList"
      >
        <div
          className="filterListItem"
          id="accepted"
          onClick={[Function]}
        >
          <img
            alt="accepted"
            id="accepted"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="filterListItem"
          id="pending"
          onClick={[Function]}
        >
          <img
            alt="pending"
            id="pending"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="filterListItem"
          id="rejected"
          onClick={[Function]}
        >
          <img
            alt="rejected"
            id="rejected"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </_default>
  </Router>
</Provider>
`;
